packages/ics721/src/ibc.rs

```diff
@@ -1,30 +1,30 @@
-use cosmwasm_schema::cw_serde;
 use cosmwasm_std::{
-    from_json, to_json_binary, Binary, DepsMut, Empty, Env, IbcBasicResponse, IbcChannelCloseMsg,
+    from_json, to_json_binary, DepsMut, Empty, Env, IbcBasicResponse, IbcChannelCloseMsg,
     IbcChannelConnectMsg, IbcChannelOpenMsg, IbcChannelOpenResponse, IbcPacket, IbcPacketAckMsg,
     IbcPacketReceiveMsg, IbcPacketTimeoutMsg, IbcReceiveResponse, Never, Reply, Response,
     StdResult, SubMsgResult, WasmMsg,
 };
 use cw_utils::parse_reply_instantiate_data;
+use ics721_types::{ibc::NonFungibleTokenPacketData, types::Ics721Status};
 use serde::{de::DeserializeOwned, Serialize};
 
 use crate::{
     helpers::ack_callback_msg,
     ibc_helpers::{ack_fail, ack_success, try_get_ack_error, validate_order_and_version},
     ibc_packet_receive::receive_ibc_packet,
     state::{
-        CLASS_ID_TO_NFT_CONTRACT, INCOMING_CLASS_TOKEN_TO_CHANNEL, NFT_CONTRACT_TO_CLASS_ID,
-        OUTGOING_CLASS_TOKEN_TO_CHANNEL, PROXY, TOKEN_METADATA,
+        CLASS_ID_TO_NFT_CONTRACT, INCOMING_CLASS_TOKEN_TO_CHANNEL, INCOMING_PROXY,
+        NFT_CONTRACT_TO_CLASS_ID, OUTGOING_CLASS_TOKEN_TO_CHANNEL, OUTGOING_PROXY, TOKEN_METADATA,
     },
-    token_types::{ClassId, TokenId},
-    types::Ics721Status,
     ContractError,
 };
 
 /// Submessage reply ID used for instantiating cw721 contracts.
 pub(crate) const INSTANTIATE_CW721_REPLY_ID: u64 = 0;
 /// Submessage reply ID used for instantiating the proxy contract.
-pub(crate) const INSTANTIATE_PROXY_REPLY_ID: u64 = 1;
+pub(crate) const INSTANTIATE_INCOMING_PROXY_REPLY_ID: u64 = 4;
```

Comment:
Why 4 instead of 2?

Reply:
scroll down, 2 and 3 is already used by `ACK_AND_DO_NOTHING` and `ACK_CALLBACK_REPLY_ID`, but since they are constants, I can rearrange it.

Reply:
cleaned it up, now ordered like this:

```rs
/// Submessage reply ID used for instantiating cw721 contracts.
pub(crate) const INSTANTIATE_CW721_REPLY_ID: u64 = 0;
/// Submessage reply ID used for instantiating the proxy contract.
pub(crate) const INSTANTIATE_INCOMING_PROXY_REPLY_ID: u64 = 1;
/// Submessage reply ID used for instantiating the proxy contract.
pub(crate) const INSTANTIATE_OUTGOING_PROXY_REPLY_ID: u64 = 2;
/// Submessages dispatched with this reply ID will set the ack on the
/// response depending on if the submessage execution succeded or
/// failed.
pub(crate) const ACK_AND_DO_NOTHING_REPLY_ID: u64 = 3;
/// Reply on callback
pub(crate) const ACK_CALLBACK_REPLY_ID: u64 = 4;
```

Reply:
Sweet, thanks for that.